my_2d_robot/src/my_2d_robo_state_publisher.cpp

```diff
@@ -0,0 +1,47 @@
+#include <string>
+#include <ros/ros.h>
+#include <sensor_msgs/JointState.h>
+#include "std_msgs/String.h"
+
+
+int main(int argc, char** argv){
+    ros::init(argc, argv, "my_2d_robo_state_publisher");
+    ros::NodeHandle n;
+    ros::Publisher joint_pub = n.advertise<sensor_msgs::JointState>("joint_states", 1000);
+    ros::Rate loop_rate(30);
+
+
+    sensor_msgs::JointState joint_state;
+    
+
+    
+    while(ros::ok){
+        joint_state.header.stamp = ros::Time::now();
+        //one message per update, This is 2 since thre are 2 joints which are updated 
+        //at the same time
+        joint_state.name = {"joint1","joint2"};
+
+        int joint1_pos = (rand()%4);
+        int joint2_pos = (rand()%4);
+
+        //position-angle of rotation
+        joint_state.position = {0,0}; //{joint1_pos,joint2_pos};
+
+        //actually publishing data
+        std_msgs::String msg;
+        ROS_INFO("%s", msg.data.c_str());
```

Comment:
We don't necessarily want to be spammed with this info on our console. Add a --debug_mode flag for the binary that only outputs this debug into to the screen when the flag is specified.